controllers/ebpf/bpfmanager-controller.go

```diff
@@ -0,0 +1,257 @@
+				},
+				InterfaceSelector: bpfmaniov1alpha1.InterfaceSelector{Interfaces: &interfaces},
+				Direction:         "egress",
+			},
+		},
+		{
+			Type: bpfmaniov1alpha1.ProgTypeTCX,
+			TCX: &bpfmaniov1alpha1.TcxProgramInfo{
+				BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+					BpfFunctionName: "tcx_ingress_pca_parse",
+				},
+				InterfaceSelector: bpfmaniov1alpha1.InterfaceSelector{Interfaces: &interfaces},
+				Direction:         "ingress",
+			},
+		},
+		{
+			Type: bpfmaniov1alpha1.ProgTypeTCX,
+			TCX: &bpfmaniov1alpha1.TcxProgramInfo{
+				BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+					BpfFunctionName: "tcx_egress_pca_parse",
+				},
+				InterfaceSelector: bpfmaniov1alpha1.InterfaceSelector{Interfaces: &interfaces},
+				Direction:         "egress",
+			},
+		},
+	}
+
+	if helper.IsFlowRTTEnabled(&fc.Spec.Agent.EBPF) {
+		bpfApp.Spec.Programs = append(bpfApp.Spec.Programs, []bpfmaniov1alpha1.BpfApplicationProgram{
+			{
+				Type: bpfmaniov1alpha1.ProgTypeFentry,
+				Fentry: &bpfmaniov1alpha1.FentryProgramInfo{
+					BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+						BpfFunctionName: "tcp_rcv_fentry",
+					},
+					FunctionName: "tcp_rcv_established",
+				},
+			},
+			{
+				Type: bpfmaniov1alpha1.ProgTypeKprobe,
+				Kprobe: &bpfmaniov1alpha1.KprobeProgramInfo{
+					BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+						BpfFunctionName: "tcp_rcv_kprobe",
+					},
+					FunctionName: "tcp_rcv_established",
+					RetProbe:     false,
+				},
+			},
+		}...)
+	}
+
+	if helper.IsNetworkEventsEnabled(&fc.Spec.Agent.EBPF) {
+		bpfApp.Spec.Programs = append(bpfApp.Spec.Programs, []bpfmaniov1alpha1.BpfApplicationProgram{
+			{
+				Type: bpfmaniov1alpha1.ProgTypeKprobe,
+				Kprobe: &bpfmaniov1alpha1.KprobeProgramInfo{
+					BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+						BpfFunctionName: "rh_network_events_monitoring",
+					},
+					FunctionName: "rh_psample_sample_packet",
+					RetProbe:     false,
+				},
+			},
+		}...)
+	}
+
+	if helper.IsPktDropEnabled(&fc.Spec.Agent.EBPF) {
+		bpfApp.Spec.Programs = append(bpfApp.Spec.Programs, []bpfmaniov1alpha1.BpfApplicationProgram{
+			{
+				Type: bpfmaniov1alpha1.ProgTypeTracepoint,
+				Tracepoint: &bpfmaniov1alpha1.TracepointProgramInfo{
+					BpfProgramCommon: bpfmaniov1alpha1.BpfProgramCommon{
+						BpfFunctionName: "kfree_skb",
+					},
+					Names: []string{"skb/kfree_skb"},
+				},
+			},
+		}...)
+	}
+}
+
+func deleteBpfApplication(ctx context.Context, c client.Client, bpfApp *bpfmaniov1alpha1.BpfApplication) error {
+	klog.Info("Deleting BpfApplication Object")
+	return c.Delete(ctx, bpfApp)
+}
+
+func createBpfApplication(ctx context.Context, c client.Client, bpfApp *bpfmaniov1alpha1.BpfApplication) error {
+	return c.Create(ctx, bpfApp)
+}
+
+func updateBpfApplication(ctx context.Context, c client.Client, bpfApp *bpfmaniov1alpha1.BpfApplication) error {
+	return c.Update(ctx, bpfApp)
+}
```

Comment:
We usually use `createdOwned`, `updateIfOwned` functions to avoid overriding object that netobserv operator didn't created or to let advanced user override a confing and skip reconciliation.

I feel that would be useful for debug purposes here